python/tank/platform/engine.py

```diff
@@ -970,6 +970,8 @@ def register_command(self, name, callback, properties=None):
                 # we know the app for the existing item.
                 # so prefix with app name
                 prefix = existing_item["properties"].get("app").instance_name
+                if existing_item["properties"].get("group"):
+                    prefix = "%s:%s" % (prefix, existing_item["properties"]["group"])
```

Comment:
Without running the code, i am not totally seeing how this will end up looking. Do you have some examples from desktop? This will affect the display name of the app right? Which i think might not be what we want. If you could add some desktop screenshots to the PR that would be great and then we should probably get @robblau's opinion!